### PR TITLE
Feature/sunset full statement version

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
## PR Overview
**This PR will address the following Issue/Feature:**
- #105 

**This PR will result in the following new package version:**

- v0.10.0 breaking because of deprecation

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## 🚨 Breaking Changes 🚨
> - The parameter `full_statement_version=false` has been fully sunset from the `sfdc_formula_view` macro. You will now need to remove this parameter to avoid a compilation error.
> 
> ## Under the hood
> - Included auto-releaser GitHub Actions workflow to automate future releases.
> - Updated maintainer pull request template.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] ~dbt run (if incremental models are present) && dbt test~

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
    
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

- Consistency tests pass
  - Screenshot 2024-06-27 at 4 16 46 PM

- When setting `full_statement_version=false` in a model, a compilation error is successfully thrown.
  - Screenshot 2024-06-27 at 12 09 08 PM

### Additional notes
- `full_statement_version=true` continues to be allowed so that users do not experience the below compilation failure since the macro cannot take an unexpected argument.
  - Screenshot 2024-06-27 at 12 04 20 PM

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: macros/sfdc_formula_view.sql
Comment (fivetran-joemarkiewicz) [inline macros/sfdc_formula_view.sql:12]: Suggestion to reword the warning slightly.

```suggestion
{{ exceptions.raise_compiler_error("\nERROR: The 'full_statement_version=false' parameter is no longer supported. Please update your " ~ this.identifier|upper ~ " model to remove this parameter.\n") }}
```

Comment (fivetran-joemarkiewicz) [inline macros/sfdc_formula_view.sql:12]: Using the `exceptions.raise_compiler_error` feels dangerous here. This means if the `full_statement_version=false` then the entire dbt project will fail with the compile error. For example, in my project I have a model that is completely unrelated to my formula_field model, but it refuses to compile because of the compile failure log.

image

I think we should definitely raise the error in this model. However, I don't advise we make this result in the entire dbt project failing.

Comment (fivetran-joemarkiewicz) [reply macros/sfdc_formula_view.sql:12]: Can we explore possible alternatives.

File: macros/sfdc_formula_view.sql
Comment (fivetran-joemarkiewicz) [inline macros/sfdc_formula_view.sql:19]: Is this comment still necessary?

File: macros/sfdc_formula_view.sql
Comment (fivetran-joemarkiewicz) [inline macros/sfdc_formula_view.sql:1]: It looks like with this update, we also no longer us the following parameters anymore:
- `reserved_table_name`
- `fields_to_include`

Can you confirm these parameters are not used? If so, we should probably document this somewhere or include some warning messages to raise in the logs that these parameters are not being used and should be removed by the end user.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-catfritz thanks for addressing this update so quickly! I have a few comments and requests before approval. 

While we are sunsetting functionality, I don't want a customer to upgrade and it possibly cause catastrophic failures in their dbt project. Ideally, we create "soft" failures and provide informative messages so customers know what actions they need to take if the release notes were not reviewed prior to upgrading.

Let me know if you want to sync on my comments and if you want to brainstorm some ideas. Thanks!

File: CHANGELOG.md
Comment (fivetran-catfritz) [reply CHANGELOG.md]: Good call. Updated!

File: macros/sfdc_formula_view.sql
Comment (fivetran-catfritz) [reply macros/sfdc_formula_view.sql]: It was actually just a result of my testing removing it and forgetting where it went. 😅 But I guess it could go last anyway since we're not really using it.

File: macros/sfdc_formula_view.sql
Comment (fivetran-catfritz) [reply macros/sfdc_formula_view.sql:1]: Ahh yes thank you. Removed!

File: macros/sfdc_formula_view.sql
Comment (fivetran-catfritz) [reply macros/sfdc_formula_view.sql]: Updated!

Comment (fivetran-catfritz) [reply macros/sfdc_formula_view.sql]: Updated with the floating string method from our discussion earlier this week!

File: macros/sfdc_formula_view.sql
Comment (fivetran-catfritz) [reply macros/sfdc_formula_view.sql]: Ahh, no. Removed.

File: macros/sfdc_formula_view.sql
Comment (fivetran-catfritz) [reply macros/sfdc_formula_view.sql]: Agree the parameters are not used. Removed and also mention them in the error.

**Review by fivetran-catfritz — COMMENTED** [review]
@fivetran-joemarkiewicz thank you for the review comments! I have made the updates, and this is ready for re-review.

File: macros/sfdc_formula_view.sql
Comment (fivetran-joemarkiewicz) [inline macros/sfdc_formula_view.sql:12]: If you want this to be on a new line you have to include the `n` after the `\`.
```suggestion
    {{ exceptions.warn("\nERROR: The full_statement_version=false, reserved_table_name, and fields_to_include parameters are no longer supported. Please update your " ~ this.identifier|upper ~ " model to remove these parameters.\n") }}
```

File: macros/sfdc_formula_view.sql
Comment (fivetran-joemarkiewicz) [inline macros/sfdc_formula_view.sql:13]: In BigQuery this error is a bit confusion.

image

What if we do this instead?
```suggestion
    See_full_model_error_in_log
```

Which provides the following error.
image

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-catfritz great work on this PR! This is nearly ready, I just have a few final comments before this is ready for approval and release.

File: README.md
Comment (fivetran-catfritz) [reply README.md]: Got it. Updated!

File: macros/sfdc_formula_view.sql
Comment (fivetran-catfritz) [reply macros/sfdc_formula_view.sql:1]: Updated!

File: macros/sfdc_formula_view.sql
Comment (fivetran-catfritz) [reply macros/sfdc_formula_view.sql]: Good catch thank you. Updated!

File: macros/sfdc_formula_view.sql
Comment (fivetran-catfritz) [reply macros/sfdc_formula_view.sql]: I see. That works for me!

**Review by fivetran-catfritz — COMMENTED** [review]
Thanks for the comments @fivetran-joemarkiewicz! I have made the updates, and it's ready for re-review!

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-catfritz just a few last comments. Let me know once ready for re-review.

File: macros/sfdc_formula_view.sql
Comment (fivetran-catfritz) [reply macros/sfdc_formula_view.sql]: From our discussion, I added the defaults for the variables back in but keeping the conditions for the warning the same.

**Review by fivetran-catfritz — COMMENTED** [review]
@fivetran-joemarkiewicz this is ready for re-review!

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:6]: Can you also include a sub bullet mentioning all of the macros that were removed as part of this update.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM Thanks for all your work on this PR @fivetran-catfritz I have just one final CHANGELOG request but that won't block the approval. Once that last CHANGELOG update is applied this will be good for release review.

File: CHANGELOG.md
Comment (fivetran-catfritz) [reply CHANGELOG.md]: Good call. Updated!

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (fivetran-jamie) [inline .github/ISSUE_TEMPLATE/bug-report.yml:2]:
```suggestion
description: Report a bug or an issue you've found within the dbt package. For SQL translation issues, please contact Fivetran [Support](https://support.fivetran.com/).
```

Comment (fivetran-jamie) [reply .github/ISSUE_TEMPLATE/bug-report.yml:2]: just in case people forget the link was on the previous page

**Review by fivetran-jamie — APPROVED** [review]
lgtm! just one suggestion to insert a link in one more place

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (fivetran-catfritz) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: Good callout. Thank you and updated!

**Pull request merged**
